hw1/matrices.py

```diff
@@ -0,0 +1,66 @@
+from math import sqrt, acos, isclose, pi
+from typing import List
```

Comment:
But hey, since this is here, let's actually add Typing to all the functions.

Reply:
It is there because I actually tried adding typings but failed :). Could you please help me a little bit with that by answering a question. What do you think the type of a vector should be? It should be something like `Iterable[Union[int, float]]` but that's too ugly. I couldn't find any other numeral types and defining my own type like `typedef number = int | float` feels wrong :)

Reply:
The first variant is actually good. With `Union` and everything. But you are right that it is weird to write it everywhere, it's long and unreadable. So try _Type aliases_ https://docs.python.org/3/library/typing.html and just create a neat alias for this type.

Reply:
Also, I found this. https://www.python.org/dev/peps/pep-0484/#the-numeric-tower

You don't need to write Union[int, float], you can just write float, then int will be autmoatically accepted.